eclipse-collections-code-generator/src/main/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGenerator.java

```diff
@@ -188,16 +196,28 @@ private ST findTemplate(String templateName)
 
     private void setTest()
     {
-        this.isTest = this.templateFile.getInstanceOf("isTest") == null
-                ? false
-                : Boolean.valueOf(this.templateFile.getInstanceOf("isTest").render());
+        this.isTest =
+                this.templateFile.getInstanceOf("isTest") != null && Boolean.valueOf(this.templateFile.getInstanceOf("isTest").render());
+    }
+
+    private void setService()
+    {
+        this.isService =
+                this.templateFile.getInstanceOf("isService") != null && Boolean.valueOf(this.templateFile.getInstanceOf("isService").render());
     }
 
     private File constructTargetPath()
     {
         ST targetPath = this.findTemplate("targetPath");
-        return this.isTest ? new File(this.moduleBaseDir, GENERATED_TEST_SOURCES_LOCATION + targetPath.render())
-                : new File(this.moduleBaseDir, GENERATED_SOURCES_LOCATION + targetPath.render());
+        if (this.isTest)
+        {
+            return new File(this.moduleBaseDir, GENERATED_TEST_SOURCES_LOCATION + targetPath.render());
+        }
+        if (this.isService)
+        {
+            return new File(this.moduleBaseDir, GENERATED_SERVICES_LOCATION + targetPath.render());
+        }
+        return new File(this.moduleBaseDir, GENERATED_SOURCES_LOCATION + targetPath.render());
```

Comment:
This if-statement makes it look like there ought to be some sort of assertion that a template is a test or service, but not both.

Reply:
This was intentional.